inc/functions.php

```diff
@@ -2908,32 +2904,18 @@ function build_clickable_smilies()
 
                     $find = str_replace(array('\\', "'"), array('\\\\', "\'"), htmlspecialchars_uni($smilie['find']));
 
-                    $onclick = " onclick=\"MyBBEditor.insertText(' $find ');\"";
-                    $extra_class = ' smilie_pointer';
-                    eval('$smilie = "'.$templates->get('smilie', 1, 0).'";');
-                    eval("\$smilie_icons .= \"".$templates->get("smilieinsert_smilie")."\";");
+                    $smilie['onclick'] = " onclick=\"MyBBEditor.insertText(' $find ');\"";
```

Comment:
Just a little bit of feedback. Is it possible to do the HTML inside the template rather than putting it in a string?

Reply:
Sure thing.